content/general/concepts/xml/terms/xpath/xpath.md

```diff
@@ -0,0 +1,116 @@
+- `position()` : Returns the numeric position of the current node.
+- `starts-with(string1, string2)` : Checks if `string1` starts with `string2` and returns true or false.
+- `sum(node-set)` : Converts the string values of the nodes in `node-set` into numbers and returns the sum of those numbers..
+- `true()` : Returns true.
+
+## Examples
+
+Consider the following XML document:
+
+```pseudo
+<A>
+  <B C="1">
+    <D E="2"/>
+  </B>
+  <B C="3">
+    <D F="4"/>
+  </B>
+</A>
+```
+
+Example XPath expressions include:
+
+- `A/B/D`, the most basic type of XPath expression. Each location step is separated by a slash. The default axis is the child axis and `A`,`B` & `C` represent the element nodes we're matching. This expression matches all `D` elements that are children of `B` elements that are children of `A` elements. It would match both `D` elements in the example. The expression can be written in full expanded form as `child::A/child::B/child::D`
+- `A//D`, also matches both `D` elements in the example, because the double slash (`//`) represents the descendant-or-self axis. This expression matches all `D` elements that are descendants of the `A` element. The expanded form of the expression is `child::A/descendant-or-self::node()/child::D`
+- `A//D[1]` adds a predicate (in the square brackets `[...]`) to the `D` node test. Using a number alone in the square brackets is a shorthand way of specifying the position of the element. This predicate specifies that we're only matching the first `D` descendant of `A`. The expanded form of the expression is `child::A/descendant-or-self::node()/child::D[position()=1]`
+- `A//@C` the `@` indicates we're searching along the attribute axis. In this case we're looking for all `C` attributes belonging to `A` or `A`'s descendants. This matches the two `C` attributes in the `B` elements in the example. The expanded form of the expression is `child::A/descendant-or-self::node()/attribute::C`
+- `//*[../@C="3"]` The `*` is a wildcard that matches any node. `..` means the parent axis, in this case the parent of the node referenced by the predicate. This expression matches any node in the document whose parent has a `C` attribute with a value of `3`. In the above case that is the second `D` element. The full expanded form of the expression is `/descenant-or-self::node()/child::node()[parent::node()/attribute::C="3"]`.
+- `//*[./@C="3"]` Is the same as the prior example, except that `.` means the self axis. In this case that refers to the node referenced by the predicate. This expression matches any node in the document that has a `C` attribute with a value of `3`. In the above case that is the second `B` element. The full expanded form of the expression is `/descenant-or-self::node()/child::node()[self::node()/attribute::C="3"]`.
```

Comment:
```suggestion
- `A/B/D`: The most basic type of XPath expression. Each location step is separated by a slash. The default axis is the child axis and `A`,`B` & `C` represent the element nodes we're matching. This expression matches all `D` elements that are children of `B` elements that are children of `A` elements. It would match both `D` elements in the example. The expression can be written in full expanded form as `child::A/child::B/child::D`
- `A//D`: Also matches both `D` elements in the example, because the double slash (`//`) represents the descendant-or-self axis. This expression matches all `D` elements that are descendants of the `A` element. The expanded form of the expression is `child::A/descendant-or-self::node()/child::D`
- `A//D[1]`: Adds a predicate (in the square brackets `[...]`) to the `D` node test. Using a number alone in the square brackets is a shorthand way of specifying the position of the element. This predicate specifies that we're only matching the first `D` descendant of `A`. The expanded form of the expression is `child::A/descendant-or-self::node()/child::D[position()=1]`
- `A//@C`: The `@` indicates we're searching along the attribute axis. In this case we're looking for all `C` attributes belonging to `A` or `A`'s descendants. This matches the two `C` attributes in the `B` elements in the example. The expanded form of the expression is `child::A/descendant-or-self::node()/attribute::C`
- `//*[../@C="3"]`: The `*` is a wildcard that matches any node. `..` means the parent axis, in this case the parent of the node referenced by the predicate. This expression matches any node in the document whose parent has a `C` attribute with a value of `3`. In the above case that is the second `D` element. The full expanded form of the expression is `/descendant-or-self::node()/child::node()[parent::node()/attribute::C="3"]`.
- `//*[./@C="3"]`: Same as the prior example, except that `.` means the self axis. In this case that refers to the node referenced by the predicate. This expression matches any node in the document that has a `C` attribute with a value of `3`. In the above case that is the second `B` element. The full expanded form of the expression is `/descendant-or-self::node()/child::node()[self::node()/attribute::C="3"]`.
```